src/main/java/com/example/demo/model/User.java

```diff
@@ -47,6 +56,12 @@ public class User implements UserDetails {
     @ToString.Exclude
     @EqualsAndHashCode.Exclude
     private Set<Role> roles;
+
+    @OneToOne(mappedBy = "user",
+            cascade = CascadeType.ALL,
+            orphanRemoval = true, fetch = FetchType.LAZY)
+    private ShoppingCart shoppingCart;
```

Comment:
I think we don't need bidirectional mapping in this case, so you simply remove it